src/dsl.jl

```diff
@@ -517,98 +488,146 @@ function extract_species_and_parameters!(reactions, excluded_syms)
         end
     end
 
-    collect(species), collect(parameters)
+    return collect(species), collect(parameters)
 end
 
-# Function called by extract_species_and_parameters!, recursively loops through an
+# Function called by extract_species_and_parameters, recursively loops through an
 # expression and find symbols (adding them to the push_symbols vector).
-function add_syms_from_expr!(push_symbols::AbstractSet, rateex::ExprValues, excluded_syms)
-    if rateex isa Symbol
-        if !(rateex in forbidden_symbols_skip) && !(rateex in excluded_syms)
-            push!(push_symbols, rateex)
+function add_syms_from_expr!(push_symbols::AbstractSet, expr::ExprValues, excluded_syms)
+    # If we have encountered a Symbol in the recursion, we can try extracting it. 
+    if expr isa Symbol
+        if !(expr in forbidden_symbols_skip) && !(expr in excluded_syms)
+            push!(push_symbols, expr)
         end
-    elseif rateex isa Expr
+    elseif expr isa Expr
         # note, this (correctly) skips $(...) expressions
-        for i in 2:length(rateex.args)
-            add_syms_from_expr!(push_symbols, rateex.args[i], excluded_syms)
+        for i in 2:length(expr.args)
+            add_syms_from_expr!(push_symbols, expr.args[i], excluded_syms)
         end
     end
-    nothing
 end
 
 ### DSL Output Expression Builders ###
 
-# Given the species that were extracted from the reactions, and the options dictionary, creates the @species ... expression for the macro output.
-function get_sexpr(species_extracted, options, key = :species;
-        iv_symbols = (DEFAULT_IV_SYM,))
+# Given the extracted species (or variables) and the option dictionary, creates the 
+# `@species ...` (or `@variables ..`) expression which would declare these.
+# If `key = :variables`, does this for variables (and not species).
+function get_usexpr(us_extracted, options, key = :species; ivs = (DEFAULT_IV_SYM,))
```

Comment:
make the names more general, as it at some point went from handling just species to species and variables (i.e. unknowns)